### PR TITLE
boost: honor `tools.build:compiler_executables` config & env vars from `[buildenv]`

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/14767

Honor `tools.build:compiler_executables` config introduced in conan 1.55.0, and properly listen env vars from profile `[buildenv]` (`self.buildenv` doesn't seem to be the correct attribute to listen).

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]
Awesome, thanks!

File: recipes/boost/all/conanfile.py [hunk 1]
```diff
@@ -1161,7 +1161,8 @@ def _ranlib(self):
 
     @property
     def _cxx(self):
-        cxx = self.buildenv.vars(self).get("CXX")
+        compilers_by_conf = self.conf.get("tools.build:compiler_executables", default={}, check_type=dict)
+        cxx = compilers_by_conf.get("cpp") or VirtualBuildEnv(self).vars().get("CXX")
```
Comment (prince-chrismc) [inline recipes/boost/all/conanfile.py:1165]: I am not sure the 2.0 stuff should move forward with env vars or us the compilers from conf 🤔 but I think we can wait to see how that recommendation falls out

This matches with the explanation I got about how this feature works

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
Hum, CI was green for 1 week, but now it's red without useful feedback?

**Comment by SpaceIm** [comment]
@prince-chrismc @jwillikers could you approve again please? Bot seems to be confused.

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**